### PR TITLE
[ENH] the final purge of cython: numba base PF, remove cython distances and remove build things

### PR DESCRIPTION
**Pull request by TonyBagnall**
#### Reference Issues/PRs
see https://github.com/alan-turing-institute/sktime/issues/1715

this does the following

1. Adjusts both versions of proximity forest to use the cython distances
2. Removes the cython distances

and hopefully
3. removes all mention/reference to/requirement for cython

### CODE REVIEW

File: docs/source/developer_guide/release.rst [hunk 1]
```diff
@@ -6,13 +6,13 @@ Releases
 This section is for core developers. To make a new release, you need
 push-to-write access on our main branch.
 
-sktime is not a pure Python package and depends on some non-Python code
-including Cython and C. We distribute compiled files, called wheels, for
+`sktime` is distributed as the so-called wheels, for
```
Comment (fkiraly) [inline docs/source/developer_guide/release.rst:9]: strange English.
"`sktime` is distributed in package builds, so-called wheels, for ..."
?

File: docs/source/installation.rst [hunk 2]
```diff
@@ -200,168 +200,6 @@ This section outlines the ``sktime`` build requirements. These are required for:
 * installing ``sktime`` from source, e.g., development versions
 * the advanced developer set-up
 
-Build requirements summary
```
Comment (fkiraly) [inline docs/source/installation.rst:203]: all the work that went into writing this section 😢

Comment (fkiraly) [reply docs/source/installation.rst:203]: good bye, old friend

File: sktime/distances/_distance.py [hunk 3]
```diff
@@ -694,15 +694,7 @@ def msm_distance(
     c: float = 0.0,
     **kwargs: Any,
 ) -> float:
-    """Compute the move-split-merge distance.
```
Comment (fkiraly) [inline sktime/distances/_distance.py:697]: why is this docstring removed? It was informative?

File: sktime/distances/_distance.py [hunk 4]
```diff
@@ -717,23 +709,6 @@ def msm_distance(
     -------
     float
         Msm distance between x and y.
-
```
Comment (fkiraly) [inline sktime/distances/_distance.py:720]: and why is this removed?

**Review by fkiraly — DISMISSED** [review]
Looks good! Finally!! GOOD RIDDANCE!!!

Just some minor comments on docstrings which you may want to fix before merge.

File: sktime/distances/base/_base.py [hunk 5]
```diff
@@ -73,12 +73,12 @@ def distance_factory(
 
         no_python_callable = self._distance_factory(x, y, **kwargs)
 
-        if not hasattr(no_python_callable, "signatures"):
-            raise RuntimeError(
-                "The distance metric specified could not be no_python"
-                "compiled. Try again and if the problem persists raise"
-                "an issue."
-            )
+        # if not hasattr(no_python_callable, "signatures"):
```
Comment (chrisholder) [inline sktime/distances/base/_base.py:76]: This needs to be commented back in

Comment (chrisholder) [reply sktime/distances/base/_base.py:76]: Remerge the msm branch and it will fix this

**Review by chrisholder — CHANGES_REQUESTED** [review]

File: docs/source/developer_guide/release.rst (hunk 1 above)
Comment (lmmentel) [reply docs/source/developer_guide/release.rst:9]: The actual line is ``sktime` is distributed as the so-called wheels, for ...` is that strange?

File: docs/source/installation.rst (hunk 2 above)
Comment (lmmentel) [reply docs/source/installation.rst:203]: :wave:

File: sktime/distances/_distance.py (hunk 3 above)
Comment (TonyBagnall) [reply sktime/distances/_distance.py:697]: Good spot, that's a merge error, I'll put it back

**Comment by TonyBagnall** [comment]
> Looks good! Finally!! GOOD RIDDANCE!!!
> 
> Just some minor comments on docstrings which you may want to fix before merge.

Yes, waiting two to three months for deprecation was tiresome but necessary I guess

File: docs/source/developer_guide/release.rst (hunk 1 above)
Comment (TonyBagnall) [reply docs/source/developer_guide/release.rst:9]: I think its ok

Comment (fkiraly) [reply docs/source/developer_guide/release.rst:9]: well, it reads a bit nebulous to someone who does not know what wheels are.
So either assume a reference frame of someone knowing this (then there´s no need for "so-called"), or assume a frame where the reader doesn´t, and give an in-line definition of what wheels are.

**Review by fkiraly — APPROVED** [review]
Looks good!

I left a minor comment regarding the "so-called wheels" which I still find confusing, but that´s not a blocker.

**Review by chrisholder — APPROVED** [review]
